src/solaris_illumos.rs

```diff
@@ -37,11 +34,9 @@ pub fn getrandom_inner(dest: &mut [u8]) -> Result<(), Error> {
         // 256 bytes is the lowest common denominator across all the Solaris
         // derived platforms for atomically obtaining random data.
         for chunk in dest.chunks_mut(256) {
-            let ret = unsafe { func(chunk.as_mut_ptr(), chunk.len(), 0) };
-            if ret != chunk.len() as _ {
-                error!("getrandom syscall failed with ret={}", ret);
-                return Err(io::Error::last_os_error().into());
-            }
+            fill_exact(chunk, |buf| unsafe {
+                func(buf.as_mut_ptr(), buf.len(), 0) as libc::ssize_t
+            })?
```

Comment:
I think `fill_exact` is redundant here.

Reply:
Ah, never mind, [docs](https://docs.oracle.com/cd/E88353_01/html/E37841/getrandom-2.html) does not specify that it's guaranteed to fill the whole buffer if it's length is less or equal to 256 bytes.